app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala

```diff
@@ -77,20 +77,20 @@ trait UpstreamConnector extends HttpErrorFunctions with Logging with UsingCircui
   ): ResponseValidation[T] =
     statusCode match {
       case NOT_FOUND => notFound(message)
-      case FORBIDDEN => logWarningAndGenDownstreamResponse(FORBIDDEN, message, BREATHING_SPACE_EXPIRED)
-      case CONFLICT => logInfoAndGenDownstreamResponse(CONFLICT, message, CONFLICTING_REQUEST)
-      case _ => logWarningAndGenDownstreamResponse(statusCode, message, SERVER_ERROR)
+      case FORBIDDEN => logAndGenDownstreamResponse(warning, FORBIDDEN, message, BREATHING_SPACE_EXPIRED)
+      case CONFLICT => logAndGenDownstreamResponse(info, CONFLICT, message, CONFLICTING_REQUEST)
+      case _ => logAndGenDownstreamResponse(warning, statusCode, message, SERVER_ERROR)
     }
 
   private def handleUpstream5xxError[T](statusCode: Int, message: String)(
     implicit r: RequestId
   ): ResponseValidation[T] =
     statusCode match {
-      case BAD_GATEWAY => logErrorAndGenDownstreamResponse(BAD_GATEWAY, message, UPSTREAM_BAD_GATEWAY)
+      case BAD_GATEWAY => logAndGenDownstreamResponse(error, BAD_GATEWAY, message, UPSTREAM_BAD_GATEWAY)
       case SERVICE_UNAVAILABLE =>
-        logErrorAndGenDownstreamResponse(SERVICE_UNAVAILABLE, message, UPSTREAM_SERVICE_UNAVAILABLE)
-      case GATEWAY_TIMEOUT => logErrorAndGenDownstreamResponse(GATEWAY_TIMEOUT, message, UPSTREAM_TIMEOUT)
-      case _ => logErrorAndGenDownstreamResponse(statusCode, message, SERVER_ERROR)
+        logAndGenDownstreamResponse(error, SERVICE_UNAVAILABLE, message, UPSTREAM_SERVICE_UNAVAILABLE)
+      case GATEWAY_TIMEOUT => logAndGenDownstreamResponse(error, GATEWAY_TIMEOUT, message, UPSTREAM_TIMEOUT)
+      case _ => logAndGenDownstreamResponse(error, statusCode, message, SERVER_ERROR)
```

Comment:
NIT: These two methods could be combined as they do the same thing really. Maybe something like: 
```
private def handleUpstreamError[T](statusCode: Int, message: String)(
    implicit r: RequestId
  ): ResponseValidation[T] = {
    val logMessage = s"Error($statusCode) for $r. $message"
    
    statusCode match {
      case NOT_FOUND => notFound(message)
      case FORBIDDEN => logAndGenDownstreamResponse(warning, logMessage, BREATHING_SPACE_EXPIRED)
      case CONFLICT => logAndGenDownstreamResponse(info, logMessage, CONFLICTING_REQUEST)
      case BAD_GATEWAY => logAndGenDownstreamResponse(error, logMessage, UPSTREAM_BAD_GATEWAY)
      case SERVICE_UNAVAILABLE => logAndGenDownstreamResponse(error, logMessage, UPSTREAM_SERVICE_UNAVAILABLE)
      case GATEWAY_TIMEOUT => logAndGenDownstreamResponse(error, logMessage, UPSTREAM_TIMEOUT)
      case _ => logAndGenDownstreamResponse(error, logMessage, SERVER_ERROR)
    }
  }
```

Reply:
I got it :smiley_cat: But as I wrote, I like keeping separated 4xx and 5xx logic.